cmd/skopeo/sync.go

```diff
@@ -536,6 +540,20 @@ func (opts *syncOptions) run(args []string, stdout io.Writer) error {
 		return err
 	}
 
+	var manifestType string
+	if opts.format.present {
+		switch opts.format.value {
+		case "oci":
+			manifestType = imgspecv1.MediaTypeImageManifest
+		case "v2s1":
+			manifestType = manifest.DockerV2Schema1SignedMediaType
+		case "v2s2":
+			manifestType = manifest.DockerV2Schema2MediaType
+		default:
+			return fmt.Errorf("unknown format %q. Choose one of the supported formats: 'oci', 'v2s1', or 'v2s2'", opts.format.value)
+		}
+	}
```

Comment:
> Do you mean that all tags in the same repo are removed on a multi-arch push? Or that somehow before there were multiple images with the same tag, and those are now cleared? Or are you just saying that after copying image1 to repo:tag and image2 to repo:tag, image1 no longer has the repo:tag label? Something else? Either way, that report is rather concerning.

When copying image1 to `repo:tag` and then copying image2 to `repo:tag`, the `tag` is removed from image1

I think this is likely a limitation of Harbor rather than any issue with Skopeo.  When I attempt to push a multi-arch image using `-all` parameter in `skopeo sync` it fails on Harbor < 2.x.  If multiple architectures for the same `image:tag` are pushed sequentially, the above-mentioned behavior is observed.

With Harbor 2.x and their support for OCI manifests, it appears to correctly infer the MIME type and uses an OCI manifest by default to support multi-arch.  


> Please share this in utils.go instead of copy&pasting from the copy subcommand.

Just to clarify, you're suggesting something like a `parseManifestFormat` function in `utils.go` to be used for `copy` and `sync` commands?

Reply:
> Just to clarify, you're suggesting something like a `parseManifestFormat` function in `utils.go` to be used for `copy` and `sync` commands?

Yes, please. (Alternatively, it could make sense to share much more of the CLI handling for setting up `copy.Options` between `copy` and `sync`, similarly to `imageOptions` providing `imageOptions.newSystemContext`, but that’s way out of scope of this PR.)

Reply:
Moved parsing to `utils.go` and added a test for it